engine/class_modules/priest/sc_priest_discipline.cpp

```diff
@@ -279,7 +279,9 @@ void priest_t::create_buffs_discipline()
 
   buffs.shadow_covenant =
       make_buff( this, "shadow_covenant", talents.discipline.shadow_covenant->effectN( 4 ).trigger() )
-          ->set_trigger_spell( talents.discipline.shadow_covenant );
+          ->set_trigger_spell( talents.discipline.shadow_covenant )
+          ->modify_duration( talents.discipline.shadow_covenant->duration() +
```

Comment:
reading the code i believe `modify_duration` _ADDS_ the input duration to base. So here you want to just add the extra amount